src/util-ja3.c

```diff
@@ -235,19 +233,15 @@ char *Ja3GenerateHash(JA3Buffer *buffer)
         return NULL;
     }
 
-    unsigned char md5[MD5_LENGTH];
-    HASH_HashBuf(HASH_AlgMD5, md5, (unsigned char *)buffer->data, buffer->used);
+    unsigned char md5[SC_MD5_LEN];
+    SCMd5HashBuffer((unsigned char *)buffer->data, buffer->used, md5, sizeof(md5));
 
     int i, x;
-    for (i = 0, x = 0; x < MD5_LENGTH; x++) {
+    for (i = 0, x = 0; x < SC_MD5_LEN; x++) {
         i += snprintf(ja3_hash + i, MD5_STRING_LENGTH - i, "%02x", md5[x]);
```

Comment:
Maybe we can enjoy this opportunity to remove this dangerous pattern `i += snprintf(buffer + i,` cf https://access.redhat.com/blogs/766093/posts/1976193

Reply:
Yeah, I have to give it another pass of where I can use the `FinalizeToHex` functions I put into the Rust hashing module. It would replace this one easily as we're only interested in the hex value as a string.

Reply:
Addressed in https://github.com/OISF/suricata/pull/5695. With the exception of the TLS fingerprint which uses a ":" separated hex string, so even with Rust we'd have to create a custom formatter for. Perhaps something to consider, but not a simple drop-in replacement.